### PR TITLE
[SYCL][Matrix] Fix bfloat16 component type matrix muladd

### PR DESCRIPTION
**Pull request by MrSidims**
Also outline MatrixOperands calculation into a separate function.

### CODE REVIEW

**Review by YuriPlyakhin — APPROVED** [review]

**Review by dkhaldi — APPROVED** [review]
LGTM

**Review by YuriPlyakhin — APPROVED** [review]
LGTM

File: sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp [hunk 1]
```diff
@@ -82,6 +82,30 @@ inline __SYCL_ALWAYS_INLINE __spv::MatrixLayout joint_matrix_layout_to_spv(
...
 }
 
+#ifdef __SPIRV_USE_COOPERATIVE_MATRIX
+template<typename Ta, typename Tb, typename Tc>
+constexpr uint32_t CalculateMatrixOperand() {
+  if constexpr (std::is_same<Ta, sycl::ext::oneapi::bfloat16>::value &&
+                std::is_same<Tb, sycl::ext::oneapi::bfloat16>::value &&
+                std::is_same<Tc, float>::value)
+    return static_cast<uint32_t>(
+        __spv::MatrixOperands::MatrixAAndBBFloat16ComponentsINTEL);
```
Comment (dkhaldi) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp:92]: why you are not using the else statement anymore?

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers can we merge this please?

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers the CI is now passing, may I ask you to merge the PR?

File: sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp [hunk 2]
```diff
@@ -82,6 +82,30 @@ inline __SYCL_ALWAYS_INLINE __spv::MatrixLayout joint_matrix_layout_to_spv(
...
+  if constexpr (std::is_unsigned<Ta>::value && std::is_signed<Tb>::value)
+    return static_cast<uint32_t>(
+        __spv::MatrixOperands::MatrixBSignedComponentsKHR);
+  if constexpr (std::is_signed<Ta>::value && std::is_signed<Tb>::value) {
+    return static_cast<uint32_t>(
+        __spv::MatrixOperands::MatrixASignedComponentsKHR) +
+           static_cast<uint32_t>(
+        __spv::MatrixOperands::MatrixBSignedComponentsKHR);
+  }
+  return 0;
```
Comment (asudarsa) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp:105]: Is this a valid return value? Or should we have some error checking here?
Thanks

Comment (MrSidims) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp:105]: Yes, it's a valid value

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers can we merge the PR?

**Comment by steffenlarsen** [comment]
@MrSidims - Would you mind pushing a merge commit. The testing results are a little old, so it would be good to make sure everything is still in order. Once that's done, this looks ready for a merge.

**Pull request merged**